src/Authentication/JWTManager.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Authentication;
+
+use CodeIgniter\I18n\Time;
+use CodeIgniter\Shield\Authentication\JWT\JWSDecoder;
+use CodeIgniter\Shield\Authentication\JWT\JWSEncoder;
+use CodeIgniter\Shield\Entities\User;
+use stdClass;
+
+/**
+ * JWT Manager
+ */
+class JWTManager
+{
+    protected Time $clock;
+    protected JWSEncoder $jwsEncoder;
+    protected JWSDecoder $jwsDecoder;
+
+    public function __construct(
+        ?Time $clock = null,
+        ?JWSEncoder $jwsEncoder = null,
+        ?JWSDecoder $jwsDecoder = null
+    ) {
+        $this->clock      = $clock ?? new Time();
+        $this->jwsEncoder = $jwsEncoder ?? new JWSEncoder(null, $this->clock);
+        $this->jwsDecoder = $jwsDecoder ?? new JWSDecoder();
+    }
+
+    /**
+     * Issues Signed JWT (JWS) for a User
+     *
+     * @param array                      $claims  The payload items.
+     * @param int|null                   $ttl     Time to live in seconds.
+     * @param string                     $keyset  The key group.
+     *                                            The array key of Config\AuthJWT::$keys.
+     * @param array<string, string>|null $headers An array with header elements to attach.
+     */
+    public function generateToken(
+        User $user,
+        array $claims = [],
+        ?int $ttl = null,
+        $keyset = 'default',
+        ?array $headers = null
+    ): string {
+        $payload = array_merge(
```

Comment:
We've used **array unpacking** before, is there a specific reason you haven't used it?

Reply:
I don't use array unpacking, because it is not fully compatible with array_merge().
We need to wait for droping PHP 8.0 and before.
https://www.php.net/releases/8.1/en.php#array_unpacking_support_for_string_keyed_arrays

Reply:
And it seems array_merge() is faster than unpacking.

The benchmark result on my mac.
```
$ php test.php 
array_merge:0.76221704483032
unpacking:1.3178927898407
```
See https://qiita.com/shigakin/items/5e4a2784bbd6b227f4d3

Reply:
Thanks, the links were interesting.